### PR TITLE
fix warning for test : UserWarning: Data provided by label_shapes don…

### PR DESCRIPTION
**Pull request by ZiyueHuang**
…'t match names specified by label_names ([] vs. ['cls_prob_label'])

### CODE REVIEW

File: example/rcnn/rcnn/symbol/symbol_resnet.py [hunk 1]
```diff
@@ -199,7 +199,7 @@ def get_resnet_test(num_classes=config.NUM_CLASSES, num_anchors=config.NUM_ANCHO
 
     # classification
     cls_score = mx.symbol.FullyConnected(name='cls_score', data=pool1, num_hidden=num_classes)
-    cls_prob = mx.symbol.SoftmaxOutput(name='cls_prob', data=cls_score)
+    cls_prob = mx.symbol.softmax(name='cls_prob', data=cls_score)
```
Comment (piiswrong) [inline example/rcnn/rcnn/symbol/symbol_resnet.py]: this is wrong. softmax is not a loss

File: example/rcnn/demo.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 LONG_SIDE = config.SCALES[0][1]
 PIXEL_MEANS = config.PIXEL_MEANS
 DATA_NAMES = ['data', 'im_info']
-LABEL_NAMES = ['cls_prob_label']
+LABEL_NAMES = None
```
Comment (piiswrong) [inline example/rcnn/demo.py]: @precedenceguo

File: example/rcnn/rcnn/symbol/symbol_resnet.py (hunk 1 above)
Comment (ZiyueHuang) [inline example/rcnn/rcnn/symbol/symbol_resnet.py]: get_resnet_test() is for test case, we only have an image as input, there is no 'cls_prob_label'. Also, TestLoader doesn't provide labels. So there is a mismatch.

Comment (ZiyueHuang) [inline example/rcnn/rcnn/symbol/symbol_resnet.py]: For test case, there is no need for SoftmaxOutput as a loss, and there is no corresponding labels.

File: example/rcnn/demo.py (hunk 2 above)
Comment (precedenceguo) [inline example/rcnn/demo.py]: It is okay because softmax does not need a label. Ziyue has tested this pr.

**Pull request merged**